sdk/storage/inc/shares/share_file_client.hpp

```diff
@@ -0,0 +1,357 @@
+     * @param options Optional parameters to execute this function.
+     * @return A UploadFilePagesResult describing the state of the updated file.
+     */
+    Azure::Core::Response<UploadFileFromResult> UploadFrom(
+        const uint8_t* buffer,
+        std::size_t bufferSize,
+        const UploadFileFromOptions& options = UploadFileFromOptions()) const;
+
+    /**
+     * @brief Creates a new file, or updates the content of an existing file. Updating
+     * an existing file overwrites any existing metadata on the file.
+     *
+     * @param file A file containing the content to upload.
+     * @param options Optional parameters to execute this function.
+     * @return A UploadFileFromResult describing the state of the updated file.
+     */
+    Azure::Core::Response<UploadFileFromResult> UploadFrom(
+        const std::string& file,
+        const UploadFileFromOptions& options = UploadFileFromOptions()) const;
+
+    /**
+     * @brief Starts copy the file specified from source URI to the file this client points to.
+     * @param copySource Specifies the URL of the source file or file, up to 2 KB in length. To copy
+     * a file to another file within the same storage account, you may use Shared Key to
+     * authenticate the source file. If you are copying a file from another storage account, or if
+     * you are copying a file from the same storage account or another storage account, then you
+     * must authenticate the source file or file using a shared access signature. If the source is a
+     * public file, no authentication is required to perform the copy operation. A file in a share
+     * snapshot can also be specified as a copy source.
+     * @param options Optional parameters to copy the content of this file.
+     * @return Azure::Core::Response<StartCopyFileResult> containing the copy related information.
+     */
+    Azure::Core::Response<StartCopyFileResult> StartCopy(
+        std::string copySource,
+        const StartCopyFileOptions& options = StartCopyFileOptions()) const;
+
+    /**
+     * @brief Aborts copying the file specified with the copy ID.
+     * @param copyId The copy identifier provided in the StartCopyFileResult of the original
+     * StartCopy operation.
+     * @param options Optional parameters to abort copying the content of this file.
+     * @return Azure::Core::Response<AbortCopyFileResult> containing the abort copy related
+     * information, current empty but preserved for future usage.
+     */
+    Azure::Core::Response<AbortCopyFileResult> AbortCopy(
+        std::string copyId,
+        const AbortCopyFileOptions& options = AbortCopyFileOptions()) const;
+
+    /**
+     * @brief Gets the properties of a file.
+     * @param options Optional parameters to get the properties of this file.
+     * @return Azure::Core::Response<GetFilePropertiesResult> containing the file properties.
+     */
+    Azure::Core::Response<GetFilePropertiesResult> GetProperties(
```

Comment:
not align with other sdk

Reply:
ditto for other APIs

Reply:
you'd better have a good reason when you decide to do something different from other Track2 sdk